backend/spec/controllers/spree/admin/search_controller_spec.rb

```diff
@@ -87,12 +87,30 @@ def starting_letters(string)
       end
     end
 
-    context 'when idds param is not present' do
+    context 'when ids param is not present' do
       let(:params) { { q: { name_cont: 'jersey' } } }
 
       it_should_behave_like 'product search' do
         let(:expected_products) { [product_one, product_two] }
       end
     end
+
+    context 'when all products are requested to be shown' do
+      context 'when a per page limit is set' do
+        let(:params) { { show_all: true, per_page: 1 } }
+
+        it_should_behave_like 'product search' do
+          let(:expected_products) { [product_one, product_two] }
+        end
+      end
+
+      context 'when a per page limit is not set' do
```

Comment:
It seems to me that this `context` expectations are the same as the `context` above, only the `params` are slightly different, so it's not evident to me what difference in result the different params should produce... so I'm missing the reason for this second test 😅. Can you help me with that?

Reply:
It is basically testing that, no matter which `per_page` you are requesting, when there's the `show_all` option it will return all the records, ignoring the `per_page` param. Makes sense @spaghetticode?

Reply:
@kennyadsl sure... thanks!